protostar/commands/test/test_command.py

```diff
@@ -131,33 +137,35 @@ async def test(
         disable_hint_validation: bool = False,
         no_progress_bar: bool = False,
         fast_collecting: bool = False,
+        safe_collecting: bool = False,
         exit_first: bool = False,
         stdout_on_success: bool = False,
     ) -> TestingSummary:
         logger = getLogger()
 
         include_paths = self._build_include_paths(cairo_path or [])
-
+        factory = (
+            StarknetPassManagerFactory
+            if safe_collecting
+            else TestCollectorPassManagerFactory
+        )
+        assert (
+            not fast_collecting
+        ), "`--fast-collecting` is deprecated, use default strategy"
```

Comment:
You can remove `fast-collecting` flag completely in this PR. The only thing left is `Command.Argument` and :::info in docs.

Reply:
It's also a logic in TestCollector and StarknetCompilation, that's why I postponed it